packages/bot/src/stores/run-panel-store.js

```diff
@@ -69,9 +71,25 @@ export default class RunPanelStore {
         }
 
         this.registerBotListeners();
+
+        const blocks_in_workspace = Blockly.derivWorkspace.getAllBlocks();
```

Comment:
@sam-binary please add a helper class in blockly and move this out. so for checking validation you must just call `Blockly_helper.isvalid()` or st like that.